### DISCUSSION TITLE
How to use a custom tiny-cuda-nn with the testbed?

### DISCUSSION THREAD

**Discussion by ben-bou** [NVlabs/instant-ngp#473] [Q&A] [answered]
Hi,

first of all, thanks for this awesome project.
I am trying to use the instant-ngp to visualize a 4D-Dataset (time, z, y, x) in the [netcdf4](https://unidata.github.io/netcdf4-python/) dataformat. For this, I built a custom tiny-cuda-nn using the PyTorch-extensions.

Now, how can I plug the trained Network into the interactive Testbed and visualize it?
Also, is there a way to deal with the time-dimension in the Testbed, or can I only visualize one time-step at once?

**Comment by Tom94** [comment, marked as answer]
Hi Ben, there's unfortunately no out-of-the-box functionality to visualize pre-trained and/or custom tiny-cuda-nn networks. Nor is instant-ngp compatible with the PyTorch bindings of tiny-cuda-nn. I'm afraid the boilerplate or a custom visualizer will have to be self-written.

**Reply by ben-bou** [reply]
Hi Tom, thanks a lot for your answer!
